tests/IssueTest.php

```diff
@@ -0,0 +1,175 @@
+            'comment_count' => 5,
+            'completed' => false,
+        ]);
+
+        $actual = json_encode($issue);
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testIssueJsonSerialize_nullableDescriptionSuccess()
+    {
+        $issue = new Issue();
+
+        $issue->id = 7;
+        $issue->title = "A Title";
+        $issue->summary = null;
+        $issue->severity = "Severe";
+        $issue->date_created = "2024-11-05 15:56:55";
+        $issue->comment_count = 5;
+        $issue->completed = false;
+
+        $result = $issue->jsonSerialize();
+
+        $this->assertNull($result['summary']);
+    }
+
+    public function testIssueJsonSerialize_nullableCommentCountSuccess()
+    {
+        $issue = new Issue();
+
+        $issue->id = 7;
+        $issue->title = "A Title";
+        $issue->summary = "Hello";
+        $issue->severity = "Severe";
+        $issue->date_created = "2024-11-05 15:56:55";
+        $issue->comment_count = null;
+        $issue->completed = false;
+
+        $result = $issue->jsonSerialize();
+
+        $this->assertNull($result['comment_count']);
+    }
+
+    public function testIssueJsonSerialize_completedTrueSuccess()
+    {
+        $issue = new Issue();
+
+        $issue->id = 7;
+        $issue->title = "A Title";
+        $issue->summary = null;
+        $issue->severity = "Severe";
+        $issue->date_created = "2024-11-05 15:56:55";
+        $issue->comment_count = 5;
+        $issue->completed = true;
+
+        $result = $issue->jsonSerialize();
+
+        $this->assertTrue($result['completed']);
+    }
+
+    public function testIssueJsonSerialize_completedFalseSuccess()
+    {
+        $issue = new Issue();
+
+        $issue->id = 7;
+        $issue->title = "A Title";
+        $issue->summary = null;
+        $issue->severity = "Severe";
+        $issue->date_created = "2024-11-05 15:56:55";
+        $issue->comment_count = 5;
+        $issue->completed = false;
+
+        $result = $issue->jsonSerialize();
+
+        $this->assertFalse($result['completed']);
+    }
+
+    public function testIssueJsonSerialize_MalformedInputs()
+    {
+        $issue = new Issue();
+
+        $this->expectException(TypeError::class);
+
+        $issue->id = "Hello";
+        $issue->title = 1;
+        $issue->summary = 2;
+        $issue->severity = 3;
+        $issue->date_created = 4;
+        $issue->comment_count = 5;
+        $issue->completed = "false";
+
+        json_encode($issue);
+
+    }
+
+    public function testGetDate_Success()
+    {
+        $date = "2024-11-05 15:56:55";
+        $expected = "05/11/2024";
+        $actual = Issue::getDate($date);
+        assertEquals($actual, $expected);
+    }
+
+    public function testGetDate_MalformedArray()
+    {
+        $date = [2024, 11, 05, 15, 56, 55];
+        $this->expectException(TypeError::class);
+        Issue::getDate($date);
+
+    }
+
+    public function testGetDate_MalformedString()
```

Comment:
```suggestion
    public function testGetDate_malformed_string()
```

This is a good malformed test because you *could* have a string that doesn't encode a real date in property.

Reply:
done